File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        if not root:
            return 0
        res = []
        def dfs(root, res):
            if not root:
                return None
            dfs(root.left, res)
            res.append(root.val)
            dfs(root.right, res)
            return res
        dfs(root, res)
        return res[k-1]

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-6.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {
    res := [][]int{}

    if root == nil{
        return res
    }

    q := []*TreeNode{root}

    for len(q) > 0{
        floor := []int{}
        qLen := len(q)
        for i := 0; i < qLen; i++{
            curr := q[0]
            q = q[1:]
            floor = append(floor, curr.Val)
            if curr.Left != nil{
                q = append(q, curr.Left)
            }
            if curr.Right != nil {
                q = append(q, curr.Right)
            }
            
        }
        res = append(res, floor)
        floor = []int{}

    }
    return res
}
